### ISSUE TITLE
Update @Column() to support enums

### ISSUE THREAD

**Issue by andyslack** [juicyllama/framework#204]

**Comment by yossi-shasho** [comment]
@andyslack I found several cases of such that on the one hand a default enum value is stated but on the other hand the field is mandatory with an `!`. Looks to me like the default value is meaningless in these cases, WDYT about removing it?
```
	@IsEnum(AppScope)
	@Column({ default: AppScope.ACCOUNT })
	scope!: AppScope
```

**Comment by andyslack** [comment]
Yes, incase there is a default option set, we should make the field optional, so this one would be:

```
@IsEnum(AppScope)
@Column({ type: 'enum', enum: AppScope, default: AppScope.ACCOUNT })
scope?: AppScope
```

Good spot!

**Comment by andyslack** [comment]
When I added defaults, it was with the intention of the fields being optional, so it should not be an issue, I prefer this than forcing data which should be optional, it will be a better developer experience to keep these optional.
